pkg/cmd/server/kubernetes/master_config.go

```diff
@@ -207,38 +201,32 @@ func BuildKubernetesMasterConfig(options configapi.MasterConfig, requestContextM
 
 	// TODO you have to know every APIGroup you're enabling or upstream will panic.  Its alternative to panicing is Fataling
 	// It needs a refactor to return errors
-	storageDestinations := genericapiserver.NewStorageDestinations()
-	// storageVersions is a map from API group to allowed versions that must be a version exposed by the REST API or it breaks.
-	// We need to fix the upstream to stop using the storage version as a preferred api version.
-	storageVersions := map[string]string{}
-
-	enabledKubeVersions := configapi.GetEnabledAPIVersionsForGroup(*options.KubernetesMasterConfig, configapi.APIGroupKube)
-	if len(enabledKubeVersions) > 0 {
-		kubeStorageVersion := unversioned.GroupVersion{Group: configapi.APIGroupKube, Version: options.EtcdStorageConfig.KubernetesStorageVersion}
-		databaseStorage, err := NewEtcdStorage(etcdClient, kubeStorageVersion, options.EtcdStorageConfig.KubernetesStoragePrefix)
-		if err != nil {
-			return nil, fmt.Errorf("Error setting up Kubernetes server storage: %v", err)
-		}
-		storageDestinations.AddAPIGroup(configapi.APIGroupKube, databaseStorage)
-		storageVersions[configapi.APIGroupKube] = options.EtcdStorageConfig.KubernetesStorageVersion
+	etcdConfig := storagebackend.Config{
```

Comment:
looks like you've lost all the cert information used to actually make the connection.